Python/errors.c

```diff
@@ -301,6 +301,23 @@ PyErr_SetString(PyObject *exception, const char *string)
     _PyErr_SetString(tstate, exception, string);
 }
 
+void
+_PyErr_SetLocaleStringTstate(PyThreadState *tstate, PyObject *exception,
+                             const char *string)
+{
+    PyObject *value = PyUnicode_DecodeLocale(string, "strict");
+    if (value != NULL) {
+        _PyErr_SetObject(tstate, exception, value);
+        Py_DECREF(value);
+    }
+}
+
+void
+_PyErr_SetLocaleString(PyObject *exception, const char *string)
+{
+    PyThreadState *tstate = _PyThreadState_GET();
+    _PyErr_SetLocaleStringTstate(tstate, exception, string);
+}
```

Comment:
It might be worth just turning this into an inline function or macro.

Reply:
Others functions are implemented without inlining and without macros so I think it's fine to keep it that way here.